### DISCUSSION TITLE
[FEATURE REQUEST] Remove "multiple state declarations"

### DISCUSSION THREAD

**Discussion by srkunze** [saltstack/salt#67458] [Feature Requests]
In our teams, we experienced confusion -- when reusing a state id -- among those who are not so experienced with salt but still need to write salt-sls-code.

So, we propose removing reusing state names completely, to simplify understanding.

We have things like:

```
state_name:
  - pkg.installed:
{% if special_condition %}
    ...
{% endif %}
    - require:
      - foo: bar
{% if special_condition2 %}
      -  ...
{% endif %}
  - service.running:
{% if special_condition3 %}
    ...
{% endif %}
    - require:
{% if special_condition4 %}
      -  ...
{% endif %}
      - foo: baz
```

Which are hard to maintain and to oversee because the second state is very much hidden among the ifs and because the "..." is larger in size people tend to forget that there's another one.

**Comment by DmitryKuzmenko** [comment]
@srkunze it sounds like better to be discussed as a [SEP](https://github.com/saltstack/salt-enhancement-proposals).
Could you please create one for this?
